engine/src/fieldhtml.cpp

```diff
@@ -1443,6 +1497,26 @@ static void import_html_append_unicode_char(import_html_t& ctxt, uint32_t p_code
 	ctxt . byte_count += 2;
 }
 
+static void import_html_append_stringref(import_html_t& ctxt, MCStringRef p_chars)
+{
+	// If the text is currently native (and there is some) or if the styling has changed
+	// then flush.
+	if ((!ctxt . is_unicode && ctxt . byte_count > 0) ||
+		!import_html_equal_style(ctxt . last_used_style, ctxt . styles[ctxt . style_index] . style))
+		import_html_flush_chars(ctxt);
+	
+	// Make sure there's enough room in the buffer.
+	if (!import_html_ensure_bytes(ctxt, sizeof(unichar_t) * MCStringGetLength(p_chars)))
+		return;
+	
+	// Append the text to the buffer.
+	ctxt . is_unicode = true;
+	MCStringGetChars(p_chars,
```

Comment:
It uses MCStringGetChars() which fetches the contents of the StringRef as UTF-16 codeunits regardless of its underlying representation.